ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/ArtifactRegistryIntegrationForm.tsx

```diff
@@ -201,27 +219,47 @@ function ArtifactRegistryIntegrationForm({
                             />
                         </FormLabelGroup>
                     )}
-                    <FormLabelGroup
-                        label="Service account key (JSON)"
-                        isRequired={values.updatePassword}
-                        fieldId="config.google.serviceAccount"
-                        touched={touched}
-                        errors={errors}
-                    >
-                        <TextArea
+                    {isCloudCredentialsEnabled && (
+                        <FormLabelGroup
+                            fieldId="config.google.wifEnabled"
+                            touched={touched}
+                            errors={errors}
+                        >
+                            <Checkbox
+                                label="Use workload identity"
+                                id="config.google.wifEnabled"
+                                isChecked={values.config.google.wifEnabled}
+                                onChange={onChange}
+                                onBlur={handleBlur}
+                                isDisabled={!isEditable}
+                            />
+                        </FormLabelGroup>
+                    )}
+                    {!values.config.google.wifEnabled && (
+                        <FormLabelGroup
+                            label="Service account key (JSON)"
                             isRequired={values.updatePassword}
-                            id="config.google.serviceAccount"
-                            value={values.config.google.serviceAccount}
-                            onChange={onChange}
-                            onBlur={handleBlur}
-                            isDisabled={!isEditable || !values.updatePassword}
-                            placeholder={
-                                values.updatePassword
-                                    ? ''
-                                    : 'Currently-stored password will be used.'
-                            }
-                        />
-                    </FormLabelGroup>
+                            fieldId="config.google.serviceAccount"
+                            touched={touched}
+                            errors={errors}
+                        >
+                            <TextArea
+                                className="json-input"
+                                isRequired={values.updatePassword}
+                                type="text"
+                                id="config.google.serviceAccount"
+                                value={values.config.google.serviceAccount}
+                                onChange={onChange}
+                                onBlur={handleBlur}
+                                isDisabled={!isEditable || !values.updatePassword}
```

Comment:
I recommend that we only disable this field when wifEnabled is true, not remove it from the DOM.
Elements suddenly appearing/disappearing  was discouraged as far by as the original 1984 Apple Macintosh Human Interface Guidelines.